src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -4600,27 +4600,34 @@ void testRebasingAnnotationChanges() throws ServiceException, InterruptedExcepti
 		String pizzaId = conceptService.create(new Concept()
 				.addDescription(new Description("Cheese pizza (food)").setTypeId(FSN))
 				.addDescription(new Description("Cheese pizza").setTypeId(SYNONYM))
+				.addAnnotation(annotation)
 				.addAxiom(new Relationship(ISA, SNOMEDCT_ROOT)), "MAIN/projectA").getConceptId();
 
 		// Create task A
 		branchService.create("MAIN/projectA/taskA");
 
-		// Add annotation to project
-		Concept pizza = conceptService.find(pizzaId, "MAIN/projectA");
-		pizza.addAnnotation(annotation);
-		conceptService.update(pizza, "MAIN/projectA");
+		branchService.create("MAIN/projectA/taskB");
+
+		// Update annotation to TaskA and promote
+		Concept pizza = conceptService.find(pizzaId, "MAIN/projectA/taskA");
+		pizza.getAnnotations().iterator().next().setAnnotationValue("Update 1");
+		conceptService.update(pizza, "MAIN/projectA/taskA");
 
 
 		// Make sure no annotation found in task A
-		pizza = conceptService.find(pizzaId, "MAIN/projectA/taskA");
-		assertTrue(pizza.getAnnotations().size() == 0);
+		pizza = conceptService.find(pizzaId, "MAIN/projectA/taskB");
+		pizza.getAnnotations().iterator().next().setAnnotationValue("Update 2");
+		conceptService.update(pizza, "MAIN/projectA/taskB");
 
-		// Rebase task A
-		branchMergeService.mergeBranchSync("MAIN/projectA", "MAIN/projectA/taskA", Collections.emptySet());
+		// Promote task A to project
+		branchMergeService.mergeBranchSync("MAIN/projectA/taskA", "MAIN/projectA", Collections.emptySet());
 
-		// Task will have one annotation after rabasing from project
-		pizza = conceptService.find(pizzaId, "MAIN/projectA/taskA");
-		assertTrue(pizza.getAnnotations().size() == 1);
+		// Rebase task B
+		MergeReview review = getMergeReviewInCurrentState("MAIN/projectA", "MAIN/projectA/taskB");
+		Collection<MergeReviewConceptVersions> conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+
+		// Expecting the annotation gets conflict
+		assertTrue(conflicts.size() != 0);
```

Comment:
@QuyenLy87 thanks for modifying this test. Can you expand it so you accept/finalise the conflict resolution? i.e. accept one of the columns and have a follow up assertion.

Reply:
Hi @dmcgihtsdo, I've updated that test case. Please have a look. Thanks

Reply:
Thanks @QuyenLy87. Do you know if the test will still pass if you change `conflict.getSourceConcept` to `concept.getAutoMergedConcept`? I am thinking an update will be needed in the `AutoMerger` class.

Reply:
Hi @dmcgihtsdo, Thanks for catching this issue. Much appreciated.
I've updated AutoMerger and the unit test. Please have a look.